server/channels/app/post.go

```diff
@@ -1338,6 +1350,10 @@ func (a *App) DeletePost(c request.CTX, postID, deleteByID string) (*model.Post,
 		a.deleteFlaggedPosts(post.Id)
 	})
 
+	if err = a.RemoveNotifications(c, post, channel, team); err != nil {
```

Comment:
@agarciamontoro Not sure if this would need load testing. New functionality is added to an API (DeletePost), and the functionality is not trivial.

My two performance concerns is that a message may have a relatively large mention list (specially for group mentions) and (if I my knowledge of Go is not so rusty) this call "blocks" the execution of the api, so any delay performing the "mention removal" would impact in a delay of the user experience of the delete action.

But 0/5.

Reply:
Everything being performed inside `RemoveNotifications` is also already being performed during create post (before this PR) as part of notification system. If its fine in create post, its absolutely fine in delete post.

I don't think we need any load testing for this.

I'll see if we can make this async and not block delete post HTTP API.

Reply:
I see this was closed. Please ping me if this work is continued

Reply:
@agarciamontoro hey, yes, please check out this PR instead - https://github.com/mattermost/mattermost/pull/23568 is the same as this PR just a new PR as this was having issue with CircleCI.